### PR TITLE
Attempt to Fix Unstable Notebooks Tests

### LINKED ISSUE #19222: Assertion Errors with Notebook Unstable Extension Unit Tests.
Notebook unstable extension unit tests are failing because of an assertion error.

### PR DESCRIPTION
**Pull request by chlafreniere**
Fixes #19222.

Attempting to fix the always-failing "unstable" tests.

Turned on exceptions, and got the `cannot read property 'getAllNotebooks' of undefined` message, indicating that `currentBook` wasn't set in this case.

Just adding a quick check against this appears to allow these tests to pass again. This should be safe because all that it does is prevent this exception from being thrown in product code.

Will continue looking at unstable tests over the next few weeks to see how unstable they really are.

### CODE REVIEW

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -709,7 +709,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	getUntitledNotebookUri(resource: string): vscode.Uri {
 		let untitledFileName = vscode.Uri.parse(`untitled:${resource}`);
-		if (!this.currentBook.getAllNotebooks().get(untitledFileName.fsPath) && !this.currentBook.getAllNotebooks().get(path.basename(untitledFileName.fsPath))) {
+		if (this.currentBook && !this.currentBook.getAllNotebooks().get(untitledFileName.fsPath) && !this.currentBook.getAllNotebooks().get(path.basename(untitledFileName.fsPath))) {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:712]: What is this even doing? A function named `get...` I wouldn't expect to have side effects like setting a path to notebook mapping.

And why are we calling basename on an untitled path anyways?

@barbaravaldez Are you familiar with this at all?

Comment (chlafreniere) [reply extensions/notebook/src/book/bookTreeView.ts:712]: Completely agree on questions re: why this function is actually adding something to a map, given its name, and I certainly don't have answers for why we're checking basename on an untitled path.

Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTreeView.ts:712]: @Charles-Gagnon I'm not familiar with this bit of code but I agree that it doesn't make sense

Comment (Charles-Gagnon) [reply extensions/notebook/src/book/bookTreeView.ts:712]: @chlafreniere Can you : 

1. Update the type of currentbook (`public currentBook: BookModel | undefined;`)
2. Look through the rest of the code and make sure it's being handled correctly elsewhere - I see a couple of places that don't do a null check

**Review by Charles-Gagnon — APPROVED** [review]

**Review by barbaravaldez — APPROVED** [review]

**Pull request merged**